### ISSUE TITLE
Content Type Controller overwrites changes made to object in controller

### ISSUE THREAD

**Issue by russdanner313** [craftercms/craftercms#6674]
### Duplicates

- [X] I have searched the existing issues

### Latest version

- [X] The issue is in the develop branch
- [X] The issue is in the latest released 4.1.x
- [ ] The issue is in the latest released 4.0.x
- [ ] The issue is in the latest released 3.1.x

### Describe the issue

One of the main use cases of the content type controller is to update properties during a write. Today, it's difficult to make changes to content because you must call the write API inside the controller, and during the RENAME operation, your content is overwritten.

If possible, maintain changes made to the dom without requiring a write during any operation (other than a delete.)

ZD: 4658

### Steps to reproduce

Steps:

1. In an Editorial Project, Go to Studio > Project tools > Content Types and edit Article
2. Edit the controller.groovy and add the content of the attached file. What this is doing is basically updating the field subject_t with the phrase "Updated on rename", only if the page is renamed or moved.
2. Create a new Article setting the page url as "article2" and the subject as "article2" as well.
3. Go and rename the article2 page, editing it and changing the page url to article3.
4. Save and check the logs you should see something like
contentLifecycleOperation >> UPDATE
contentLifecycleOperation >> RENAME
Subject before update article2
Subject after update Updated on rename
5. Edit again the page and you will see the subject is still the old one (article2).

controller:
```
import scripts.libs.CommonLifecycleApi
import org.craftercms.studio.impl.v1.util.XmlUtils;
import org.dom4j.Element;
import org.dom4j.Node

println "contentLifecycleOperation >> " + contentLifecycleOperation

def document = contentLoader.getContent(site, path)
def root = document.getRootElement();

if ("RENAME" == contentLifecycleOperation) {

    def subjectNode = root.selectSingleNode("//subject_t")
    println "Subject before update "+subjectNode.text
    subjectNode.setText("Updated on rename")
    def contentService = applicationContext.getBean("cstudioContentService");
    def contentStream = new ByteArrayInputStream((XmlUtils.convertDocumentToString(document)).getBytes("UTF-8"));
    contentService.writeContent(site, path, contentStream);
    println "Subject after update "+subjectNode.text
}

def contentLifecycleParams =[:]
contentLifecycleParams.site = site
contentLifecycleParams.path = path
contentLifecycleParams.user = user
contentLifecycleParams.contentType = contentType
contentLifecycleParams.contentLifecycleOperation = contentLifecycleOperation
contentLifecycleParams.contentLoader = contentLoader
contentLifecycleParams.applicationContext = applicationContext

def controller = new CommonLifecycleApi(contentLifecycleParams)
controller.execute()
```

**Comment by phuongnq** [comment]
`RENAME` is a bit special case. In the implementation, we perform an additional write after the content has been moved:

```
// This write is performed after processing the commitIds so we don't miss any commit
 if (movedDocument != null) {
    writeContent(site, movePath, ContentUtils.convertDocumentToStream(movedDocument, CONTENT_ENCODING));
}               
```
I moved this write before we run `updateDatabaseOnMove` because in the `updateDatabaseOnMove`, we trigger the lifecycle event RENAME. Updating content along with the RENAME event will be lost if the `writeContent` is run after this.
cc @sumerjabri @jmendeza
